tools/cli/commands/changelog.js

```diff
@@ -291,7 +291,9 @@ async function getProjectChangeTypes( needChangelog ) {
 	const types = {};
 	for ( const proj of needChangelog ) {
 		const composerJSON = readComposerJson( proj );
-		if (
+		if ( ! composerJSON ) {
+			process.exit( 1 );
+		} else if (
 			composerJSON.extra &&
 			composerJSON.extra.changelogger &&
 			composerJSON.extra.changelogger.types
```

Comment:
I think this case would probably be safer to do like
```suggestion
		if ( composerJSON?.extra?.changelogger?.types
```
instead of exiting here in the middle of trying to process a list.

Ideally calling code should have already validated the project list anyway, where it can make a better decision on "warn about the bad ones and continue with any good ones" versus "error out".